### PR TITLE
Add julia to deployment

### LINKED ISSUE #200: Fix Julia package build
#199 is failing during build. Check why the build is failing on BB5 and fix the recipe.

### PR DESCRIPTION
**Pull request by pramodk**
Fixes #200
Closes #199

### CODE REVIEW

**Comment by pramodk** [comment]
@matz-e : could you take a look here when you have some time?

**Comment by matz-e** [comment]
@pramodk that's a very non-descript failure to me.

Retest this please

**Comment by pramodk** [comment]
@matz-e : I think we have to build this locally on BB5 and track the issue

**Comment by matz-e** [comment]
Did we have a JIRA ticket for this?

**Comment by matz-e** [comment]
Investigation so far from [discourse](https://discourse.julialang.org/t/source-build-julia-0-6-0-pre-alpha-branch-fails-with-error-in-inference-ji/10689/5)

> Set MARCH=haswell in Make.user solved the issue.
> There is another post that states the same issue, I am pasting the answer here:
> “So this happens to be an issue with LLVM on Skylake CPUs. It should be fixed on master but meanwhile you can work around the issue by setting MARCH=haswell when building Julia.”

**Comment by matz-e** [comment]
@pramodk why do we use 0.6.2? Shouldn't we use 1.0 or 1.1? This version here is EOL: [see the announcement](https://discourse.julialang.org/t/julia-0-6-and-0-7-are-now-officially-unmaintained/20449)

**Comment by pramodk** [comment]
* yes, there is a ticket related to Julia from Polina (see Julia integration with NEURON)
* if newer version builds, we should use it.

> from my tiny mobile screen.

**Comment by matz-e** [comment]
Please retest.

Wondering if that's a transient failure… Github not being reachable.

**Comment by matz-e** [comment]
Please retest

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Comment by matz-e** [comment]
Retest this please

**Review by matz-e — APPROVED** [review]
I approve!

@pramodk any last words before I merge this? NGV-103 tests have been successful.

File: deploy/configs/serial-libraries/modules.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ modules:
       - gmsh
       - hdf5~mpi
       - highfive~mpi
+      - julia-dev
```
Comment (matz-e) [inline deploy/configs/serial-libraries/modules.yaml]:
```suggestion
```

**Comment by pramodk** [comment]
good to merge! I missed your last comment.

**Pull request merged**